bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/discovery/SatelDeviceDiscoveryService.java

```diff
@@ -152,14 +152,10 @@ private void addDevice(DeviceType deviceType, int deviceKind, int deviceId, Stri
 
     private void addThing(ThingTypeUID thingTypeUID, @Nullable String deviceId, String label,
             Map<String, Object> properties) {
-        ThingUID bridgeUID = bridgeHandler.getThing().getUID();
-        ThingUID thingUID;
-        if (deviceId == null) {
-            thingUID = new ThingUID(thingTypeUID, bridgeUID.getId());
-        } else {
-            thingUID = new ThingUID(thingTypeUID, bridgeUID, deviceId);
-        }
-        DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withThingType(thingTypeUID)
+        final ThingUID bridgeUID = bridgeHandler.getThing().getUID();
+        final ThingUID thingUID = new ThingUID(thingTypeUID, bridgeUID,
+                deviceId == null ? toCamelCase(thingTypeUID.getId()) : deviceId);
```

Comment:
If `deviceId` is `null` the new thing will get the id based on the `thingTypeUID`. This is the same for all instances of a specific type. This means there can only be 1 per type. I don't know if that is intended, but it does look strange.

Reply:
Yes, that's intended. The old code was creating in such case thing UID without device ID, for the same purpose. But in OH3 this is not possible. Basically these thing types are "virtual" things, they logically group channels for specific purpose, for example accessing event log of connected alarm system. All these channels could be implemented directly in the bridge, but I decided to make it this way. Maybe this wasn't good idea, but I don't want to make a revolution now, just to fix the issue with UID.